src/viur/core/modules/file.py

```diff
@@ -449,6 +493,53 @@ def hmac_sign(data: t.Any) -> str:
     def hmac_verify(data: t.Any, signature: str) -> bool:
         return hmac.compare_digest(File.hmac_sign(data.encode("ASCII")), signature)
 
+    @staticmethod
+    def create_serve_parameters(serving_url: str) -> t.Iterable[str] | None:
+        """
+        Splits a serving URL into its components, used by serve function.
+        :param serving_url: the serving URL to be split
+        :return: return a tuple of host and key
+        """
+        regex_result = re.match(
+            r"^https:\/\/(.*?)\.googleusercontent\.com\/(.*?)$",
+            serving_url
+        )
+
+        if regex_result:
+            return regex_result.groups()
+        return regex_result
+
+    @staticmethod
+    def create_serve_url(
+        serving_url: str,
```

Comment:
Can we make the method a little bit more handy?

So that I only have to pass it a file skel (`SkeletonInstance` for a `FileSkel`) or dest object (`SkeletonInstance` for a `RefSkelForFile`), and it will automatically fill in the `servingurl` and `filename` from it.

So I either pass a servingurl or fileskel?!

Reply:
Either you create a proposal for the function, or we use it now as is, in the first iteration. This is a big feature and nice that we have a solution for it right now, so lets get things merged. v3.7 is still not Final.